src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfToken.java

```diff
@@ -0,0 +1,152 @@
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.security.authtoken.jwt.JwtVendor;
+import org.opensearch.security.securityconf.ConfigModel;
+import org.opensearch.security.securityconf.DynamicConfigModel;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.user.User;
+import org.opensearch.threadpool.ThreadPool;
+
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class CreateOnBehalfOfToken extends BaseRestHandler {
+
+    private JwtVendor vendor;
+    private final ThreadPool threadPool;
+
+    private ConfigModel configModel;
+
+    private DynamicConfigModel dcm;
+
+    @Subscribe
+    public void onConfigModelChanged(ConfigModel configModel) {
+        this.configModel = configModel;
+    }
+
+    @Subscribe
+    public void onDynamicConfigModelChanged(DynamicConfigModel dcm) {
+        this.dcm = dcm;
+        if (dcm.getDynamicOnBehalfOfSettings().get("signing_key") != null && dcm.getDynamicOnBehalfOfSettings().get("encryption_key") != null) {
+            this.vendor = new JwtVendor(dcm.getDynamicOnBehalfOfSettings(), Optional.empty());
+        } else {
+            this.vendor = null;
+        }
+    }
+
+    public CreateOnBehalfOfToken(final Settings settings, final ThreadPool threadPool) {
+        this.threadPool = threadPool;
+    }
+
+    @Override
+    public String getName() {
+        return getClass().getSimpleName();
+    }
+
+    @Override
+    public List<Route> routes() {
+        return addRoutesPrefix(
+                ImmutableList.of(
+                        new Route(Method.POST, "/user/onbehalfof")
+                )
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        switch (request.method()) {
+            case POST:
+                return handlePost(request, client);
+            default:
+                throw new IllegalArgumentException(request.method() + " not supported");
+        }
+    }
+
+    private RestChannelConsumer handlePost(RestRequest request, NodeClient client) throws IOException {
+        return new RestChannelConsumer() {
+            @Override
+            public void accept(RestChannel channel) throws Exception {
+                final XContentBuilder builder = channel.newBuilder();
+                BytesRestResponse response;
+                try {
+                    if (vendor == null) {
+                        channel.sendResponse(new BytesRestResponse(RestStatus.SERVICE_UNAVAILABLE, "on_behalf_of configuration is not being configured"));
```

Comment:
Nit: Is this message correct? I thought this was for calling the token creation not for configuring the Security settings of the token creator?

Reply:
Yes, so basically this line is handling the case that when a user doesn't have the `on_behalf_of` section configured in the security's config and still trying to call the obo endpoint. According to this scenario, Craig also suggested that we should also provide some useful error msg to show user what is going on, and that's why I put something like "Obo setting is not there, so this service is not available" Otherwise, it will just go through the config file and give user an error like:
```
{
  "error": {
    "root_cause": [
      {
        "type": "json_generation_exception",
        "reason": "Can not start an object, expecting field name (context: Object)"
      }
    ],
    "type": "json_generation_exception",
    "reason": "Can not start an object, expecting field name (context: Object)"
  },
```
Which is also make sense, but it may confuse the user to understand what is going on.